src/Microsoft.AspNet.Razor.Runtime/Resources.resx

```diff
@@ -189,4 +189,10 @@
   <data name="TagHelperAttributeList_CannotAddAttribute" xml:space="preserve">
     <value>Cannot add a {0} with inconsistent names. The {1} property '{2}' must match the location '{3}'.</value>
   </data>
+  <data name="TagHelperDescriptorFactory_InvalidRestrictChildrenAttributeName" xml:space="preserve">
+    <value>Invalid child restriction '{0}' for tag helper '{1}'. Tag helpers cannot restrict child elements that contain a '{2}' character.</value>
+  </data>
+  <data name="TagHelperDescriptorFactory_InvalidRestrictChildrenAttributeNameNullWhitespace" xml:space="preserve">
+    <value>Invalid child restriction tag name for tag helper '{0}'. Name cannot be null or whitespace.</value>
```

Comment:
For both of these, suggest referencing the `RestrictChildrenAttribute` type and its property rather than "child restriction". Generally make these resources more like `TagHelperDescriptorFactory_InvalidAttributePrefixNotNull` and the other invalid XYZ resources in this file.